Assets/SEE/Controls/MarkAction.cs

```diff
@@ -0,0 +1,162 @@
+using System.Collections.Generic;
+using SEE.GO;
+using SEE.Net.Actions;
+using SEE.Utils.History;
+using UnityEngine;
+using SEE.Audio;
+using SEE.Game.SceneManipulation;
+using SEE.Utils;
+
+namespace SEE.Controls.Actions
+{
+    /// <summary>
+    /// Action to create a new node for a selected city.
+    /// </summary>
+    internal class MarkAction : AbstractPlayerAction
+    {
+        /// <summary>
+        /// If the user clicks with the mouse hitting a game object representing a graph node,
+        /// this graph node is a parent to which a new node is created and added as a child.
+        /// <see cref="IReversibleAction.Update"/>.
+        /// </summary>
+        /// <returns>true if completed</returns>
+        public override bool Update()
+        {
+            bool result = false;
+
+            // FIXME: Needs adaptation for VR where no mouse is available.
+            if (Input.GetMouseButtonDown(0)
+                && Raycasting.RaycastGraphElement(out RaycastHit raycastHit, out GraphElementRef _) == HitGraphElement.Node)
+            {
+                // the hit object is the parent in which to create the new node
+                GameObject parent = raycastHit.collider.gameObject;
+                Vector3 position = parent.transform.position;
+                Vector3 scale = parent.transform.lossyScale;
+                addedSphere = GameNodeMarker.AddSphere(parent, position: position, worldSpaceScale: scale);

```

Comment:
Why passing position and scale as parameters? Couldn't `GameNodeMarker.AddSphere` retrieve these?